### PR TITLE
Enable extension without reloading

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

This PR updates extension manager to be able to enable voting reputation extension without reloading page (after installing)

**New stuff** ✨

* reinitializeColonyManager util saga

**Changes** 🏗

* refactored setupUserContext to use new saga
* refactored colonyExtensionEnable saga to clear client which we want to enable from extensionClients and set whenever it will be enabled

Resolves DEV-393

### CODE REVIEW

**Review by chinins — COMMENTED** [review]
The installing & enabling without refreshing works fine. But when I then create a motion, I am redirected to a not found page. I then referesh it and the motion details page appears fine. I wonder if it's linked to your PR. It doesn't happen on the `feature/reputation-voting` branch

**Comment by alicjakujawa** [comment]
> The installing & enabling without refreshing works fine. But when I then create a motion, I am redirected to a not found page. I then referesh it and the motion details page appears fine. I wonder if it's linked to your PR. It doesn't happen on the `feature/reputation-voting` branch

Can be! good catch i will investigate it :)

File: src/modules/dashboard/sagas/extensions.ts [hunk 1]
```diff
@@ -126,10 +131,17 @@ function* colonyExtensionEnable({
...
   const initChannelName = `${meta.id}-initialise`;
   const setPermissionChannelName = `${meta.id}-setUserRoles`;
+  const colonyManager = TEMP_getContext(ContextModule.ColonyManager);
 
   if (!extension) {
     throw new Error(`Extension with id ${extensionId} does not exist!`);
   }
+
+  const key = `${colonyAddress}-${extensionId}`;
+  // Remove old exentions client if exist
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/sagas/extensions.ts:141]: Typo on comment `extensions`

**Review by ArmandoGraterol — APPROVED** [review]
It seems to be working as advertised :ok_hand:

**Review by rdig — APPROVED** [review]
Tested and works very nicely.

A very nice refactor which fixes both the Extensions as well as the clients rehydration.

Way to go :100:

**Review by chinins — APPROVED** [review]
Working well now! Great job!

**Pull request merged**